tests/integration/test_coredns.py

```diff
@@ -29,41 +27,55 @@ async def test_core_dns(e: OpsTest):
         )
         await e.model.wait_for_idle(["microk8s"])
 
-    u: Unit = e.model.applications["microk8s"].units[0]
+    yield e.model.applications["microk8s"]
+
 
+@pytest.fixture()
+async def coredns(e: OpsTest, microk8s):
```

Comment:
deploy the `coredns` charm as an `active/idle` fixture and the model with the offers available to link the machine model. 

on teardown -- remove the offer and the application